sources/pmwebapi_test.go

```diff
@@ -54,7 +54,76 @@ func TestUnmarshal(t *testing.T) {
 		t.Fatalf("Not receiving correct URL. Expected: %s, Got: %s", expected, testMap.URL)
 	}
 }
+func TestWeirdMetrics(t *testing.T) {
```

Comment:
This could probably be much cleaner and more efficient. 90% of the code here is repeated multiple times, and the remaining 10% can easily be set as variables. I would recommend creating an array of the tested and expected results then iterating through them. This function could be about 10 lines long, while still easily adding new scenarios without copying yourself.

Reply:
Also, as mentioned before, remember to test _other_ scenarios as well - ones that would potentially break this. That's what the point of unit tests is about really. Remember, you always expect a new car to work in perfect weather, but how does it handle and perform when it's snowing, raining, freezing, or in sweltering heat?

Reply:
Also also, add a newline before this function - currently it runs into the previous one.